### ISSUE TITLE
Annotation Processor Artifact Is of Type POM

### ISSUE THREAD

**Issue by gilday** [ryandens/auto-delegate#3]
The `auto-delegate-processor` artifact's descriptor in Maven Central says it is of type `pom`, but the intent is for it to be a `jar`. 

Maven users can workaround this by providing the `all` classifier to retrieve the jar produced by the shadow plugin:

```xml
<plugin>
    <artifactId>maven-compiler-plugin</artifactId>
    <configuration>
        <annotationProcessorPaths>
            <path>
                <groupId>com.ryandens</groupId>
                <artifactId>auto-delegate-processor</artifactId>
                <version>${auto-delegate.version}</version>
                <classifier>all</classifier>
            </path>
        </annotationProcessorPaths>
    </configuration>
</plugin>
```

The same workaround does not work for Gradle users:

```
Execution failed for task ':auto-delegate-examples:compileJava'.
> Could not resolve all files for configuration ':auto-delegate-examples:annotationProcessor'.
   > Could not resolve com.ryandens:auto-delegate-processor:0.2.0.
     Required by:
         project :auto-delegate-examples
      > Project :auto-delegate-examples declares a dependency from configuration 'annotationProcessor' to configuration 'all' which is not declared in the descriptor for com.ryandens:auto-delegate-processor:0.2.0.
```

Recommend forgoing the shadow plugin for a library like this.

**Comment by ryandens** [comment]
You're the worst :laughing: but also 99% correct as usual.

I don't agree with your solution of not using the shadow plugin. This is recommended best practice from the Google auto project (issue describing why they do this [here](https://github.com/google/auto/issues/105)) but basically, it's common for annotation processors to use similar dependencies and this can cause weird conflicts. In order to be compatible with other annotation processors, it's recommended to shade processor dependencies as they're only used at compile time and don't impact the size of the resulting application artifact. 

That being said, you're definitely right that the artifact descriptor is not correct. I misinterpreted how nebula wants to interact with the shadow jar plugin (I think I read [this comment](https://github.com/nebula-plugins/nebula-publishing-plugin/blob/master/src/main/groovy/nebula/plugin/publishing/maven/MavenShadowPublishPlugin.groovy#L26) wrong) so I disabled the jar task and caused this bad behavior. Enabling the jar task fixes this. I have this setup up in PR #4 if you want to take a look and verify this will solve your use case.

I am surprised though, I swear I tried this out on one of our old repos after I published it because it came up on a PR discussion.
